### PR TITLE
feat(trial balance): add extra checks

### LINKED ISSUE #1713: design: extra Trial Balance rules
Our Trial Balance currently catches SQL errors that might crop up, and a few logical errors with dates in the wrong period, etc.

There are several rules that we have not properly handled and documented:
 1. Locked Debtor Groups
 2. Locked Creditor Groups
 3. Locked/Deactivated Users
 4. ~~Locked Fiscal Years (See #1647)~~
 5. ~~Locked Periods~~

These cases should be clearly handled by the Trial Balance module on the server, and the errors displayed  clearly to the client.  We might also add a way for the user to understand what actions they must take to correct these errors.

### PR DESCRIPTION
**Pull request by jniles**
This commit adds extra trial balance checks to make sure no locked users, creditor groups, or debtor groups are used in the Trial Balance before posting.

Closes #1713.

### CODE REVIEW

File: client/src/i18n/fr/posting_journal.json [hunk 1]
```diff
@@ -26,31 +26,34 @@
...
-      "DEB_CRED_NOT_NULL" : "Le débit et le crédit ne doivent pas être égale à Zero",
       "MISSING_DATES" : "Date(s) manquante(s)",
+      "MISSING_DESCRIPTION" : "Descriptions omises",
       "MISSING_DOCUMENT_ID" : "Ligne(s) sans ID Document",
       "MISSING_ENTITY" : "Ligne(s) sans ID entite",
       "MISSING_FISCAL_OR_PERIOD" : "Année fiscale ou période omise",
+      "MISSING_TRANSACTIONS" : "Aucune transaction n'a été soumise.",
       "MULTIPLE_CANCELLING" : "Annulation multiple. Cette transaction a deja été annulée",
+      "NO_FISCAL_FOR_DATE" : "Il n'existe pas d'année fiscale pour cette Date",
       "SINGLE_LINE_TRANSACTION" : "Transaction(s) avec une ligne",
```
Comment (lomamech) [inline client/src/i18n/fr/posting_journal.json]: Try to use "Une transaction doit comporter deux lignes ou plus. Pour corriger cela, localisez la transaction et ajoutez la ligne d'équilibrage." for key "SINGLE_LINE_TRANSACTION"

Comment (jniles) [reply client/src/i18n/fr/posting_journal.json]: Merci!

File: server/models/procedures/trial_balance.sql [hunk 2]
```diff
@@ -130,6 +130,22 @@ BEGIN
     JOIN period p ON pj.period_id = p.id
     WHERE p.locked = 1 GROUP BY pj.record_uuid;
 
+  -- check that there are no transactions with accounts of locked creditor groups
+  INSERT INTO stage_trial_balance_errors
+    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.LOCKED_CREDITOR_GROUP_ACCOUNT' AS code
+    FROM posting_journal AS pj JOIN stage_trial_balance_transaction AS temp
```
Comment (lomamech) [inline server/models/procedures/trial_balance.sql:136]: I just tested it locally and it works very well, but I would like to know why the temporary table stage_trial_balance_transaction was created

**Comment by jniles** [comment]
bors ping

**Pull request merged**